kyo-core/shared/src/main/scala/kyo/meters.scala

```diff
@@ -86,47 +86,45 @@ object Meters:
     ): Meter < (IOs & S1 & S2 & S3 & S4) =
         pipeline[S1 & S2 & S3 & S4](List(m1, m2, m3, m4))
 
-    def pipeline[S](l: Seq[Meter < (IOs & S)]): Meter < (IOs & S) =
-        Seqs.collect(l).map { meters =>
+    def pipeline[S](meters: Seq[Meter < (IOs & S)]): Meter < (IOs & S) =
+        Seqs.collect(meters).map { seq =>
+            val meters = seq.toIndexedSeq
             new Meter:
 
                 val available =
-                    def loop(l: Seq[Meter], acc: Int): Int < IOs =
-                        l match
-                            case Seq() => acc
-                            case h +: t =>
-                                h.available.map(v => loop(t, acc + v))
-                    loop(meters, 0)
-                end available
+                    Loops.indexed(0) { (idx, acc) =>
+                        if idx == meters.length then Loops.done(acc)
+                        else meters(idx).available.map(v => Loops.continue(acc + v))
+                    }
 
                 def run[T, S](v: => T < S) =
```

Comment:
`run` and `tryRun` aren't stack safe because it's necessary to produce a thunk representing the execution of all other nested meters to handle permits correctly. That's not problematic unless for some reason there's a pipeline with hundreds of meters, which doesn't seem to make sense.